models/articles.model.js

```diff
@@ -1,5 +1,39 @@
 const db = require("../db/connection");
 
+exports.fetchArticles = (topic) => {
+  if (topic) {
+    return db
+      .query(
+        `SELECT articles.*, CAST (COUNT(comment_id) AS INT) AS comment_count 
+      FROM articles 
+      JOIN comments ON articles.article_id = comments.article_id WHERE topic = $1 GROUP BY articles.article_id ORDER BY article_id ASC;`,
+        [topic]
+      )
+      .then((data) => {
+        if (!data.rows.length) {
+          return Promise.reject({ status: 404, message: "Article not found" });
+        }
+        data.rows.forEach((index) => {
```

Comment:
We can specifically request the columns from the table that we would like back within the SQL query. Right now you are selecting everything from articles on like 7. We could specify specific columns and negate the need to run this forEach.